src/main/java/com/tjtanjin/steve/commands/CommandHandler.java

```diff
@@ -0,0 +1,138 @@
+package com.tjtanjin.steve.commands;
+
+import java.util.HashMap;
+
+import com.tjtanjin.steve.tasks.TaskHandler;
+
+/**
+ * The CommandHandler class manages all supported commands and provides them with the
+ * TaskHandler to modify tasks when necessary.
+ */
+public class CommandHandler {
+
+    private final ByeCommand BYE_COMMAND;
+    private final DeadlineCommand DEADLINE_COMMAND;
+    private final DeleteCommand DELETE_COMMAND;
+    private final DoneCommand DONE_COMMAND;
+    private final EventCommand EVENT_COMMAND;
+    private final FindCommand FIND_COMMAND;
+    private final HelpCommand HELP_COMMAND;
+    private final ListCommand LIST_COMMAND;
+    private final TodoCommand TODO_COMMAND;
+
+    /**
+     * Constructor for CommandHandler.
+     *
+     * @param taskHandler task handler to pass to commands that modify tasks
+     */
+    public CommandHandler(TaskHandler taskHandler) {
+
+        //load command descriptions
+        //list storing commands/descriptions, ideally store in json file
+        HashMap<String, String> cmdInfo = new HashMap<>();
+        cmdInfo.put("BYE", "bye | Description: exits the program");
+        cmdInfo.put("LIST", "list | Description: list all entered tasks");
+        cmdInfo.put("DONE",
+                "done <task index> | Description: marks by index a given task as done");
+        cmdInfo.put("TODO", "todo <name> | Description: adds a new todo task");
+        cmdInfo.put("DEADLINE",
+                "deadline <name> /by <end date> | Description: adds a new deadline task");
+        cmdInfo.put("EVENT",
+                "event <name> /from <start date> /to <end date> | Description: adds a new event task");
+        cmdInfo.put("DELETE", "delete <task index> | Description: delete by index a given task");
```

Comment:
Perhaps change the description from "delete..." to "deletes..." for consistency with the rest of your descriptions.

Reply:
Fixed, thank you! :)